lib/services/kube_cluster_resource_wrapper.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"github.com/gravitational/teleport/api/types"
+)
+
+// KubernetesClusterRBAC is a minimal implementation of
+// services.AccessCheckable for *types.KubernetesCluster.
+// This allows us to use common RBAC checks for Kubernetes Clusters
+// and other resources.
+type KubernetesClusterRBAC struct {
+	namespace string
+	cluster   *types.KubernetesCluster
+}
+
+const kindKubeClusterRBAC = "kube_cluster"
+
+func NewKubernetesClusterRBAC(namespace string, kube *types.KubernetesCluster) KubernetesClusterRBAC {
```

Comment:
The proper way would be to make a proper `KubernetesCluster` resource. This would also pave the way (somewhat) for separating it into a separate resource in future (similar to apps and databases).

Reply:
Correct. Andrew advised against going down that path for now and revisiting in the future as it's a larger refactor and the goal of this is just to pave the way for desktop access, not rewrite Kubernetes access.

Reply:
This feels quite hacky TBH. I would spend a little extra time and add a proper KubernetesCluster resource. You don't need to refactor a whole lot, just define a resource and a conversion method from old struct (which is a part of the Server) to the resource. I did the same for Apps.

Reply:
Defined the new resource and conversion method. Let me know what you think.